CDP4Grapher/Behaviors/GrapherOrgChartBehavior.cs

```diff
@@ -0,0 +1,328 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="GrapherOrgChartBehavior.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft,
+
+//
+//    This file is part of CDP4-IME Community Edition. 
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Grapher.Behaviors
+{
+    using System;
+    using System.Linq;
+    using System.Windows;
+
+    using CDP4Common.EngineeringModelData;
+
+    using CDP4Grapher.Utilities;
+    using CDP4Grapher.ViewModels;
+
+    using DevExpress.Diagram.Core;
+    using DevExpress.Diagram.Core.Layout;
+    using DevExpress.Xpf.Diagram;
+    using DevExpress.XtraPrinting;
+    using DevExpress.XtraRichEdit.Model;
+
+    using Direction = DevExpress.Diagram.Core.Direction;
+
+    /// <summary>
+    /// Allows proper callbacks on the diagramming tool
+    /// </summary>
+    public class GrapherOrgChartBehavior : DiagramOrgChartBehavior, IGrapherOrgChartBehavior
+    {
+        /// <summary>
+        /// Gets or sets the current layout
+        /// </summary>
+        public (LayoutEnumeration layout, Enum direction) CurrentLayout { get; private set; } = (LayoutEnumeration.TipOver, TipOverDirection.LeftToRight);
+
+        /// <summary>
+        /// Holds the value whether the <see cref="DiagramControl"/> has loaded for the first time
+        /// </summary>
+        private bool hasLoaded;
+
+        /// <summary>
+        /// The on attached event handler
+        /// </summary>
+        protected override void OnAttached()
```

Comment:
i would expect to see a OnDetaching as well to lean up the event handlers

Reply:
It's here on line 183